odoo/addons/base/models/ir_ui_view.py

```diff
@@ -614,20 +619,35 @@ def apply_inheritance_specs(self, source, specs_tree, inherit_id):
                 elif pos == 'inside':
                     add_text_inside(node, spec.text)
                     for child in spec:
-                        node.append(child)
+                        new_node = child
+                        if child.get('position') == 'move':
+                            to_move = self.locate_node(source, child)
+                            new_node = copy.deepcopy(to_move)
+                            remove_element(to_move)
+                        node.append(new_node)
                 elif pos == 'after':
                     # add a sentinel element right after node, insert content of
                     # spec before the sentinel, then remove the sentinel element
                     sentinel = E.sentinel()
                     node.addnext(sentinel)
                     add_text_before(sentinel, spec.text)
                     for child in spec:
-                        sentinel.addprevious(child)
+                        new_node = child
+                        if child.get('position') == 'move':
+                            to_move = self.locate_node(source, child)
+                            new_node = copy.deepcopy(to_move)
+                            remove_element(to_move)
+                        sentinel.addprevious(new_node)
                     remove_element(sentinel)
                 elif pos == 'before':
                     add_text_before(node, spec.text)
                     for child in spec:
-                        node.addprevious(child)
+                        new_node = child
+                        if child.get('position') == 'move':
+                            to_move = self.locate_node(source, child)
+                            new_node = copy.deepcopy(to_move)
+                            remove_element(to_move)
```

Comment:
```python
def get_target(target):
    if child.get('position') == 'move':
        to_move = self.locate_node(source, child)
        target = copy.deepcopy(to_move)
        remove_element(to_move)
    return child
```
as an inner function, what do you think?

Reply:
Yes, indeed, fixed :smiley:

Reply:
Updated accordingly, thanks.

Reply:
While we're bikeshedding, `get_target()` sounds like a confusing name in this context, as the "target" is firstly the destination node, and here it's applied to the source node.

Reply:
`get_target()` also implies that you're merely retrieving it, while this function alters the document tree. Like a `read()` override that would call `write()` ;-)

Reply:
I have updated the PR with a clearer version (`_move_node`).